# Calculate format=diff merge(media-libs/allegro)=>5.2.4,5.2.5
commit fd29999446c0d29177b6902443c3732057cf4049
Author: Christian Mauduit (ufoot) <ufoot@ufoot.org>
Date:   Sun Dec 23 09:10:27 2018 +0100

    Added ifdef to prevent hard ref to glXCreateContextAttribsARB

diff --git a/src/x/xglx_config.c b/src/x/xglx_config.c
index eebe2a0a7..ab9636914 100644
--- a/src/x/xglx_config.c
+++ b/src/x/xglx_config.c
@@ -491,8 +491,12 @@ static GLXContext create_context_new(int ver, Display *dpy, GLXFBConfig fb,
    GCCA_PROC _xglx_glXCreateContextAttribsARB = NULL;
 
    if (ver >= 140) {
-      /* GLX 1.4 should have this */
+      /* GLX 1.4 should have this, if it's defined, use it directly. */
+      /* OTOH it *could* be there but only available through dynamic loading. */
+      /* In that case, fallback to calling glxXGetProcAddress. */
+#ifdef glXCreateContextAttribsARB
       _xglx_glXCreateContextAttribsARB = glXCreateContextAttribsARB;
+#endif // glXCreateContextAttribsARB
    }
    if (!_xglx_glXCreateContextAttribsARB) {
       /* Load the extension manually. */
